Rotational fluids (stationary)

SeMardt opened the pull request:
stationary test (rotational couette flow) implemented and running

schmaikos committed:
Update ns_problem.py

Added Coriolis and Euler Force, additional dimensionless parameter Rossby number.

schmaikos committed:
Update ns_bdf_solver.py

Added cases for Coriolis and Euler force with Rossby number.

schmaikos committed:
Update ns_solver_base.py

Added Coriolis and Euler force wirh Rossby number.

schmaikos committed:
Update ns_solver_base.py

Not an Expression (?)

schmaikos committed:
Update ns_solver_base.py

Not an Expression. (?)

schmaikos committed:
Create forces_in_rotational_system

Setup for Coriolis and Euler force.

schmaikos committed:
Create test_stationary_rotational_system.py

Test

schmaikos committed:
Rename forces_in_rotational_system to forces_in_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

SeMardt committed:
Update test_stationary_rotational_system.py

added Ro, changed bcs

SeMardt committed:
Update test_stationary_rotational_system.py

SeMardt committed:
Update test_stationary_rotational_system.py

added forces for first testing

schmaikos committed:
Update forces_in_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

Need to replace setup for the coriolis and euler force into solver base (dependency of sol_v)

schmaikos committed:
Update ns_bdf_solver.py

Find out the type of Spatial.Coordinate(self._mesh)

schmaikos committed:
Update ns_solver_base.py

assert isinstance(u, self._form_function_types), Find equivalent for spatial coordinate.

Omega and Alpha is not None needs to be reworked, empty dlfn.Constant or set it to 0

modify time for alpha and omega?

schmaikos committed:
Update ns_problem.py

file name with omega and alpha?
set_parameters problem with alpha and omega

schmaikos committed:
Update test_stationary_rotational_system.py

Input for omega and alpha needs to be reviewed

schmaikos committed:
Delete forces_in_rotational_system.py

Dont need it anymore

schmaikos committed:
Update test_stationary_rotational_system.py

schmaikos committed:
Update ns_solver_base.py

schmaikos committed:
Update ns_bdf_solver.py

schmaikos committed:
Update ns_problem.py

schmaikos committed:
Update ns_solver_base.py

SeMardt committed:
Added rotational couette flow test

Exterior bc must still be adjusted

SeMardt committed:
Update test_stationary_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

schmaikos committed:
Update test_stationary_rotational_system.py

SeMardt committed:
Update grid_generator.py

changed inner boundary of the spherical shell to the right radius

SeMardt committed:
Update test_stationary_rotational_system.py

modified rotational couette, should work properly now

SeMardt committed:
Update test_stationary_rotational_system.py

removed rotational cavity flow due to impracticality

sebglane commented on source/ns_solver_base.py:
```diff
@@ -231,6 +231,29 @@ def _viscous_term(self, u, v):
+        assert isinstance(self._Alpha, dlfn.Constant)
+        if self._mesh.geometry().dim() == 2:
+            assert len(self._Alpha.ufl_shape) == 0
+            return dot(dlfn.as_vector((-self._Alpha * dlfn.SpatialCoordinate(self._mesh)[1], self._Alpha * dlfn.SpatialCoordinate(self._mesh)[0])), v)  
```
This line is not readable. Please define a variable `x = dlfn.SpatialCoordinate(self._mesh)`.

sebglane reviewed (changes requested):
Minor change, see comment.

sebglane commented:
You need to merge the `main` branch.

schmaikos committed:
Update ns_solver_base.py

schmaikos committed:
Update ns_bdf_solver.py

Commit:
Merge branch 'main' into rotational_fluids

Commit:
Merge branch 'main' into rotational_fluids

Commit:
Merge branch 'main' into rotational_fluids

Commit:
Reduced resolution of mesh generated by grid_generator.spherical_shell because of out of memory error

Commit:
Added pragma no cover statements

sebglane commented:
Please check my modifications. I changed quite a lot. Most important is that the code reproduces the results which you have obtained previously.

schmaikos committed:
Add files via upload

sebglane commented on tests/test_instationary_rotational_couette.py:
```diff
@@ -0,0 +1,78 @@
+        omega_start = 0
+        self._alpha_start = 1
+        self._acceleration_time = 1.0
+        self.set_parameters(Re=550.0, Ro=1.0, Omega=dlfn.Constant(omega_start), Alpha=dlfn.Constant(self._alpha_start))
```
This test failes because the method `set_parameters` does not exist anymore. I have modified the code structure and the handling of dimensionless parameters recently. Please have a look at the other tests how the parameters should be specified now.

schmaikos committed:
Update ns_problem.py

Added angular_velocity functions and in while loop set_time for AngularVelocityVector

schmaikos committed:
Add files via upload

schmaikos committed:
Rename test_instationary_rotational_systems.py to test_instationary_rotational_system.py

schmaikos committed:
Delete test_instationary_rotational_couette.py

schmaikos committed:
Update grid_generator.py

Changed refinements to points

schmaikos committed:
Update test_stationary_rotational_system.py

Changed from refinements to points

schmaikos committed:
Update test_instationary_rotational_system.py

Removed print

schmaikos committed:
Update test_instationary_rotational_system.py

schmaikos committed:
Update test_instationary_rotational_system.py

schmaikos committed:
Update ns_problem.py

schmaikos committed:
Update test_instationary_rotational_system.py

Commit:
Merge branch 'main' into rotational_fluids

Commit:
Fixed errors, removed unnecessary code, fixed PEP8 warnings
